<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>914</width>
    <height>599</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* ========== 全局样式 ========== */
QMainWindow {
    background-color: #f5f7fa; /* 主窗口浅灰背景 */
}

/* ========== 按钮样式 ========== */
QPushButton {
    background-color: #3498db; /* 主色调：蓝色 */
    color: white;              /* 文字白色 */
    border: none;              /* 无边框 */
    border-radius: 6px;        /* 圆角6px */
    padding: 6px 16px;         /* 内边距：上下6px，左右16px */
    font-size: 8pt;           /* 字体大小 */
}
/* 鼠标悬浮 */
QPushButton:hover {
    background-color: #2980b9; /* 深一点的蓝色 */
}
/* 鼠标按下 */
QPushButton:pressed {
    background-color: #1f618d; /* 更深的蓝色 */
}
/* 禁用状态 */
QPushButton:disabled {
    background-color: #bdc3c7;
    color: #7f8c8d;
}
/* 清空图表按钮特殊样式（红色） */
QPushButton#btn_clearChart {
    background-color: #e74c3c;
}
QPushButton#btn_clearChart:hover {
    background-color: #c0392b;
}
/* 统计按钮特殊样式（绿色） */
QPushButton#btn_stat {
    background-color: #2ecc71;
}
QPushButton#btn_stat:hover {
    background-color: #27ae60;
}

/* ========== 输入框/下拉框/数字框/日期框 ========== */
QLineEdit, QComboBox, QSpinBox, QDateEdit {
    border: 1px solid #ddd;    /* 浅灰边框 */
    border-radius: 4px;        /* 圆角4px */
    padding: 4px;              /* 内边距 */
    font-size: 8pt;           /* 字体大小 */
    background-color: white;   /* 白色背景 */
}
/* 聚焦时样式（蓝色边框） */
QLineEdit:focus, QComboBox:focus, QSpinBox:focus, QDateEdit:focus {
    border-color: #3498db;
    outline: none;             /* 无默认外边框 */
}

/* ========== 下拉框弹出列表 ========== */
QComboBox QAbstractItemView {
    border: 1px solid #ddd;
    border-radius: 4px;
    background-color: white;
    selection-background-color: #3498db; /* 选中行蓝色 */
    selection-color: white;              /* 选中行文字白色 */
}

/* ========== 表格样式 ========== */
QTableView {
    border: 1px solid #ddd;
    border-radius: 6px;        /* 表格圆角 */
    background-color: white;
    alternate-row-color: #f9f9f9; /* 隔行浅灰 */
    gridline-color: #eee;       /* 网格线浅灰 */
}
/* 表格表头 */
QTableView::horizontalHeader {
    background-color: #3498db;
    color: white;
    font-weight: bold;
    border: none;
}
QTableView::horizontalHeader::section {
    padding: 8px;
    border: none;
}
/* 表格垂直表头 */
QTableView::verticalHeader {
    background-color: #f5f7fa;
    border: none;
}
/* 表格选中行 */
QTableView::item:selected {
    background-color: #3498db;
    color: white;
}

/* ========== 分组框（GroupBox） ========== */
QGroupBox {
    font-weight: bold;         /* 标题加粗 */
    border: 1px solid #ddd;    /* 边框 */
    border-radius: 8px;        /* 圆角8px */
    margin-top: 10px;          /* 顶部外边距 */
    padding-top: 10px;         /* 顶部内边距 */
    background-color: white;   /* 白色背景 */
}
/* 分组框标题 */
QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;                /* 标题左移10px */
    top: -3px;                 /* 标题上移7px（悬浮效果） */
    padding: 0 8px;            /* 标题左右内边距 */
    background-color: #f5f7fa; /* 标题背景与主窗口一致 */
}

/* ========== 文本编辑框（统计区） ========== */
QTextEdit {
    border: 1px solid #ddd;
    border-radius: 6px;
    background-color: white;
    padding: 8px;
    font-size: 8pt;
}

/* ========== 标签样式 ========== */
QLabel {
    font-size: 8pt;
    color: #2c3e50; /* 深灰文字（更清晰） */
}

/* ========== 图表容器 ========== */
QWidget#widget_chart {
    border: 1px solid #ddd;
    border-radius: 8px;
    background-color: white;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" rowspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>250</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>成绩数据录入</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>学生姓名：</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="le_name"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>班级：</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="le_class"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>课程：</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="le_course"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>成绩：</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="sb_score">
           <property name="maximum">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>考试时间：</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDateEdit" name="dateEdit">
           <property name="displayFormat">
            <string>yyyy-MM-dd</string>
           </property>
           <property name="calendarPopup">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="btn_add">
           <property name="text">
            <string>添加成绩</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_delete">
           <property name="text">
            <string>删除选中</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_clear">
           <property name="text">
            <string>清空输入</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" colspan="3">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>100</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>班级/课程筛选</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>261</width>
         <height>83</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>班级：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="cb_class"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>课程：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="cb_course"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="4" rowspan="3">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QTextEdit" name="te_stat">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_chart" native="true">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>200</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>83</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QPushButton" name="btn_stat">
        <property name="text">
         <string>成绩统计分析</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_trend">
        <property name="text">
         <string>成绩趋势折线图</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_chart">
        <property name="text">
         <string>成绩占比饼图</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_clearChart">
        <property name="text">
         <string>清除图表</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_export">
        <property name="text">
         <string>导出成绩报表</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="3">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>82</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="0" colspan="4">
     <widget class="QTableView" name="tableView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <property name="sortingEnabled">
       <bool>true</bool>
      </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
